tests/aldn1000.py

```diff
@@ -0,0 +1,199 @@
+import unittest
+
+from parameterized import parameterized
+
+from utils.channel_access import ChannelAccess
+from utils.ioc_launcher import get_default_ioc_dir
+from utils.test_modes import TestModes
+from utils.testing import get_running_lewis_and_ioc, skip_if_recsim
+
+
+DEVICE_PREFIX = "ALDN1000_01"
+DEVICE_NAME = "aldn1000"
+
+IOCS = [
+    {
+        "name": DEVICE_PREFIX,
+        "directory": get_default_ioc_dir("ALDN1000"),
+        "macros": {},
+        "emulator": DEVICE_NAME,
+    },
+]
+
+
+TEST_MODES = [TestModes.RECSIM, TestModes.DEVSIM]
+
+
+class Aldn1000Tests(unittest.TestCase):
+    """
+    Tests for the Aldn1000 IOC.
+    """
+    def setUp(self):
+        self._lewis, self._ioc = get_running_lewis_and_ioc(DEVICE_NAME, DEVICE_PREFIX)
+        self.ca = ChannelAccess(device_prefix=DEVICE_PREFIX)
+        self._lewis.backdoor_run_function_on_device("reset")
+
+    @parameterized.expand([('Value 1', 12.12), ('Value 2', 1.123), ('Value 3', 123.0)])
+    @skip_if_recsim("Not supported in RECSIM")
+    def test_GIVEN_new_diameter_WHEN_set_diameter_THEN_new_diameter_set(self, _, value):
+        expected_diameter = value
+        self.ca.set_pv_value("DIAMETER:SP", expected_diameter)
+
+        self.ca.assert_that_pv_is("DIAMETER", expected_diameter, timeout=2)
+
+    @parameterized.expand([('Value 1', 12345), ('Value 2', 1234), ('Value 3', 77424)])
+    @skip_if_recsim("Not supported in RECSIM")
+    def test_GIVEN_new_invalid_high_diameter_WHEN_set_diameter_THEN_diameter_set_limit_returned(self, _, value):
+        invalid_diameter = value
+        expected_diameter = 1000.00
+        self.ca.set_pv_value("DIAMETER:SP", invalid_diameter)
+
+        self.ca.assert_that_pv_is("DIAMETER", expected_diameter, timeout=2)
+
+    @parameterized.expand([('Value 1', -2345), ('Value 2', -1234), ('Value 3', -0676424)])
```

Comment:
The superfluous 0 here isn't supported for python 3

Reply:
Removed typo.